content/intro-to-storybook/react-native/en/composite-component.md

```diff
@@ -18,303 +18,311 @@ Since `Task` data can be sent asynchronously, we **also** need a loading state t
-import { task, actions } from './Task.stories';
-import TaskList from './TaskList';
-
-export const defaultTasks = [
-  { ...task, id: '1', title: 'Task 1' },
-  { ...task, id: '2', title: 'Task 2' },
-  { ...task, id: '3', title: 'Task 3' },
-  { ...task, id: '4', title: 'Task 4' },
-  { ...task, id: '5', title: 'Task 5' },
-  { ...task, id: '6', title: 'Task 6' },
-];
-export const withPinnedTasks = [
-  ...defaultTasks.slice(0, 5),
-  { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
-];
-
-storiesOf('TaskList', module)
-  .addDecorator(story => <View style={[styles.TaskBox, { padding: 42 }]}>{story()}</View>)
-  .add('default', () => <TaskList tasks={defaultTasks} {...actions} />)
-  .add('withPinnedTasks', () => <TaskList tasks={withPinnedTasks} {...actions} />)
-  .add('loading', () => <TaskList loading tasks={[]} {...actions} />)
-  .add('empty', () => <TaskList tasks={[]} {...actions} />);
+```jsx:title=components/TaskList.stories.jsx
+import { TaskList } from "./TaskList";
+import { Default as TaskStory } from "./Task.stories";
+import { View } from "react-native";
+
+export default {
+  component: TaskList,
+  title: "TaskList",
+  decorators: [
+    (Story) => (
+      <View style={{ padding: 42, backgroundColor: "#26c6da", flex: 1 }}>
+        <Story />
+      </View>
+    ),
+  ],
+};
+
+export const Default = {
+  args: {
+    // Shaping the stories through args composition.
+    // The data was inherited from the Default story in Task.stories.js.
+    tasks: [
+      { ...TaskStory.args.task, id: "1", title: "Task 1" },
+      { ...TaskStory.args.task, id: "2", title: "Task 2" },
+      { ...TaskStory.args.task, id: "3", title: "Task 3" },
+      { ...TaskStory.args.task, id: "4", title: "Task 4" },
+      { ...TaskStory.args.task, id: "5", title: "Task 5" },
+      { ...TaskStory.args.task, id: "6", title: "Task 6" },
+    ],
+  },
+};
+
+export const WithPinnedTasks = {
+  args: {
+    // Shaping the stories through args composition.
+    // Inherited data coming from the Default story.
+    tasks: [
+      ...Default.args.tasks.slice(0, 5),
+      { id: "6", title: "Task 6 (pinned)", state: "TASK_PINNED" },
+    ],
+  },
+};
+
+export const Loading = {
+  args: {
+    tasks: [],
+    loading: true,
+  },
+};
+
+export const Empty = {
+  args: {
+    // Shaping the stories through args composition.
+    // Inherited data coming from the Loading story.
+    ...Loading.args,
+    loading: false,
+  },
+};
 ```
 
-As you may have noticed, the `addDecorator()` was used in the previous chapter and in this one, it allows us to add some “context” to the rendering of each task. In this case we add padding around the list to make it easier to visually verify.
+As you may have noticed, a decorator was used in the previous chapter and in this one, it allows us to add some “context” to the rendering of each task. In this case we add padding around the list to make it easier to visually verify.
 
```

Comment:
@dannyhw reframe this to include decorators per our conversation. We're not mentioning them in the "Simple Component" section", as we don't have a React Native version of the documentation; point at the React version of the documentation.

Reply:
Updated it